cmd/vicadmin/server.go

```diff
@@ -46,10 +55,78 @@ const (
 	formatZip
 )
 
+const sessionExpiration = time.Hour * 24
+const sessionCookieKey = "sessiondata"
+
+// UserSession holds a user's session metadata
+type UserSession struct {
+	username string
+	created  time.Time
+	config   *session.Config
+}
+
+// UserSessionStore holds and manages user sessions
+type UserSessionStore struct {
+	*sync.Mutex
+	sessions map[string]*UserSession
+	ticker   *time.Ticker
+}
+
+func (u *UserSessionStore) Add(username string, config *session.Config) *UserSession {
+	u.Lock()
+	defer u.Unlock()
+	u.sessions[username] = &UserSession{
+		username: username,
+		created:  time.Now(),
+		config:   config,
+	}
+	return u.sessions[username]
+}
+
+func (u *UserSessionStore) Delete(username string) {
+	u.Lock()
+	defer u.Unlock()
+	delete(u.sessions, username)
+}
+
+func (u *UserSessionStore) Get(username string) *UserSession {
+	return u.sessions[username]
+}
+
+// GetSession logs into vSphere and returns a session object. Caller responsible for error handling/logout
+func (u *UserSessionStore) GetSession(username string) (vSphereSession *session.Session, err error) {
+	return vSphereSessionGet(u.Get(username).config)
+}
+
+// reaper takes abandoned sessions to a farm upstate so they don't build up forever
+func (u *UserSessionStore) reaper() {
+	select {
+	case <-u.ticker.C:
```

Comment:
Don't understand why we need this. Just the for loop should suffice? Or should this be in a loop? See also https://golang.org/pkg/time/#Tick